addons/mail/static/tests/models/attachment_card_qunit_tests.js

```diff
@@ -0,0 +1,16 @@
+/** @odoo-module **/
+
+import { patchRecordMethods } from '@mail/model/model_core';
+// ensure that the model definition is loaded before the patch
+import '@mail/models/attachment_card';
+
+patchRecordMethods('AttachmentCard', {
+    /**
+     * @override
+     */
+    onClickUnlink(value) {
+        const { messagingBus } = this.messaging;
+        this._super(value);
+        messagingBus.trigger('attachment_deleted');
+    },
+});
```

Comment:
There are other flows that might lead to attachments being deleted, this patch is a bit too weak, I would directly put in correct place in business code.

Reply:
Ideally should tell which attachment is deleted and properly compare in test.